spyder/utils/introspection/numpy_docstr.py

```diff
@@ -106,8 +108,8 @@ def _search_return_in_numpydocstr(docstr):
     return found
 
 
-@memoize_default(None, evaluator_is_first_arg=True)
-def find_return_types(evaluator, func):
+#@memoize_default()
```

Comment:
I left it there to show that I disabled caching for that function to make monkey patch work.

I had to change the way I'm monkey patching and I couldn't get the caching to work.  Luckily, I didn't notice a difference in the run speed.

I am able to cache with the upstream patch.

I can give you more details on the problem later today.